code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,371 @@
+							"/obj/item/cocktail_stuff/celery")
+
+	var/possible_wedges = list("/obj/item/reagent_containers/food/snacks/plant/orange/wedge", \
+	                        "/obj/item/reagent_containers/food/snacks/plant/lime/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/lemon/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/grapefruit/wedge")
+
+	New()
+		..()
+		// Start by getting a few initial things
+		if (!src.home || !isturf(src.home))
+			src.home = get_turf(src)
+		if (src.homeTables.len == 0)
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables |= T
+
+	proc/get_empty_tables()
+		if (src.homeTables.len == 0)
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables |= T
+			if (src.homeTables.len == 0)
+				explode()
+		for (var/obj/table/reinforced/bar/T in src.homeTables)
+			if (!istype(T))
+				src.homeTables -= T
+			var/glasses = 0
+			for (var/obj/item/reagent_containers/food/drinks/drinkingglass in view(0, get_turf(T)))
+				glasses++
+			if (glasses < 3)
+				src.targets |= T
+
+	process()
+		// Nothing to do. Let's find something to do.
+		if (targets.len == 0)
+			get_empty_tables()
+			if (targets.len == 0) // No work to be done, let's go home.
+				if (!isturf(home))
+					explode()
+					return
+				if (get_turf(src) == get_turf(home)) return
+				src.navigate_to(get_turf(home), FLOORBOT_MOVE_SPEED, max_dist = 10)
+				if (!src.path || !length(src.path))
+					KillPathAndGiveUp(1)
+					return
+				return
+		// Something to do. Let's decide what to do first.
+		if (!target)
+			target = targets[1]
+		// Now we need to find out if we're going to move towards a prop dispenser or the table.
+		if (!moveTowards)
+			if (!hasDrink)
+				// if there's a barbuddy dispenser nearby, let's do the cute little animation thing. if not, use magic to summon a drink
+				for (var/obj/decal/fakeobjects/barbuddy_dispenser/D in view(5, src))
+					moveTowards = D
+				if (!moveTowards)
+					hasDrink = 1
+			else
+				moveTowards = targets[1]
+
+		if (IN_RANGE(get_turf(src), get_turf(src.moveTowards), 1))
+			bartend()
+			return
+
+		if (isnull(get_turf(moveTowards)))
+			if (moveTowards in src.homeTables)
+				src.homeTables -= moveTowards
+			KillPathAndGiveUp(1)
+			return
+
+		if (!src.path || !length(src.path))
+			src.navigate_to(get_turf(moveTowards), FLOORBOT_MOVE_SPEED, max_dist = 10)
+			if (!src.path || !length(src.path))
+				KillPathAndGiveUp(1)
+				return
+
+	KillPathAndGiveUp(var/give_up)
+		. = ..()
+		if(give_up)
+			src.targets -= src.target
+			src.target = null
+			moveTowards = null
+			// Let's check if we've been stolen.
+			if (!(src.home in view(5, src)))
+				explode()
+
+	proc/bartend()
+		if (istype(moveTowards, /obj/decal/fakeobjects/barbuddy_dispenser)) // If it's the dispenser, do a little animation.
+			playsound(moveTowards.loc, 'sound/misc/pourdrink2.ogg', 50, 1, 0.3)
+			moveTowards.icon_state = "alc_dispenser[rand(1,5)]"
```

Comment:
Haven't checked to see what the dispenser looks like, but if its an animation, you might want to use something like flick(), since it (I think) removes the image after it animates. Note, I've never used it.